fog/view/server/tests/router_smoke_tests.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use futures::executor::block_on;
+use mc_common::logger::{create_app_logger, log, o, Logger};
+use mc_crypto_keys::{CompressedRistrettoPublic, RistrettoPublic};
+use mc_fog_kex_rng::KexRngPubkey;
+use mc_fog_recovery_db_iface::{RecoveryDb, ReportData, ReportDb};
+use mc_fog_test_infra::db_tests::random_block;
+use mc_fog_view_server_test_utils::RouterTestEnvironment;
+use mc_util_from_random::FromRandom;
+use rand::{rngs::StdRng, SeedableRng};
+use std::{thread::sleep, time::Duration};
+
+async fn test_router_integration(test_environment: &mut RouterTestEnvironment, logger: Logger) {
+    let store_count = 5;
+    let mut rng: StdRng = SeedableRng::from_seed([123u8; 32]);
+    let db = test_environment
+        .db_test_context
+        .as_ref()
+        .unwrap()
+        .get_db_instance();
+
+    let ingress_key = CompressedRistrettoPublic::from(RistrettoPublic::from_random(&mut rng));
+    let accepted_block_1 = db.new_ingress_key(&ingress_key, 0).unwrap();
+    assert_eq!(accepted_block_1, 0);
+
+    db.set_report(
+        &ingress_key,
+        "",
+        &ReportData {
+            pubkey_expiry: 6,
+            ingest_invocation_id: None,
+            report: Default::default(),
+        },
+    )
+    .unwrap();
+
+    let total_block_count = store_count;
+    let egress_public_key = KexRngPubkey {
+        public_key: [1; 32].to_vec(),
+        version: 0,
+    };
+
+    let invoc_id1 = db
+        .new_ingest_invocation(None, &ingress_key, &egress_public_key, 0)
+        .unwrap();
+
+    let mut expected_records = Vec::new();
+    for i in 0..total_block_count {
```

Comment:
:wrench: It feels like there is an unnecessary level of indirection for `total_block_count`.
At the top of the function there is `store_count = 5`.  It's never used until 11 lines up where there is `let total_block_count = store_count`, and then `total_block_count` is only ever used here.
Suggest doing something like
```suggestion
    let block_count = 5
    for i in 0..block_count {
```
I'm not sure that `total` is needed in this use case here.

Reply:
Done. Made it a constant too.